server/queue/queue-api/src/test/java/org/apache/james/queue/api/ManageableMailQueueContract.java

```diff
@@ -0,0 +1,642 @@
+            .mimeMessage(createMimeMessage())
+            .sender(MailAddressFixture.ANY_AT_JAMES)
+            .recipients(MailAddressFixture.OTHER_AT_LOCAL, MailAddressFixture.OTHER_AT_JAMES)
+            .lastUpdated(new Date())
+            .build());
+
+        getManageableMailQueue().remove(ManageableMailQueue.Type.Sender, MailAddressFixture.OTHER_AT_LOCAL.asString());
+
+        assertThat(getManageableMailQueue().browse())
+            .extracting(ManageableMailQueue.MailQueueItemView::getMail)
+            .extracting(Mail::getName)
+            .containsExactly(NAME2);
+    }
+
+    @Test
+    default void removeByRecipientShouldRemoveSpecificEmail() throws Exception {
+        getManageableMailQueue().enQueue(FakeMail.builder()
+            .name(NAME1)
+            .mimeMessage(createMimeMessage())
+            .sender(MailAddressFixture.OTHER_AT_LOCAL)
+            .recipients(MailAddressFixture.ANY_AT_JAMES)
+            .lastUpdated(new Date())
+            .build());
+        getManageableMailQueue().enQueue(FakeMail.builder()
+            .name(NAME2)
+            .mimeMessage(createMimeMessage())
+            .sender(MailAddressFixture.ANY_AT_JAMES)
+            .recipients(MailAddressFixture.OTHER_AT_LOCAL)
+            .lastUpdated(new Date())
+            .build());
+
+        getManageableMailQueue().remove(ManageableMailQueue.Type.Recipient, MailAddressFixture.OTHER_AT_LOCAL.asString());
+
+        assertThat(getManageableMailQueue().browse())
+            .extracting(ManageableMailQueue.MailQueueItemView::getMail)
+            .extracting(Mail::getName)
+            .containsExactly(NAME1);
+    }
+
+    @Test
+    default void removeByRecipientShouldRemoveSpecificEmailWhenMultipleRecipients() throws Exception {
+        getManageableMailQueue().enQueue(FakeMail.builder()
+            .name(NAME1)
+            .mimeMessage(createMimeMessage())
+            .sender(MailAddressFixture.OTHER_AT_LOCAL)
+            .recipients(MailAddressFixture.ANY_AT_JAMES, MailAddressFixture.OTHER_AT_JAMES)
+            .lastUpdated(new Date())
+            .build());
+        getManageableMailQueue().enQueue(FakeMail.builder()
+            .name(NAME2)
+            .mimeMessage(createMimeMessage())
+            .sender(MailAddressFixture.ANY_AT_JAMES)
+            .recipients(MailAddressFixture.OTHER_AT_LOCAL, MailAddressFixture.OTHER_AT_JAMES)
+            .lastUpdated(new Date())
+            .build());
+
+        getManageableMailQueue().remove(ManageableMailQueue.Type.Recipient, MailAddressFixture.OTHER_AT_LOCAL.asString());
+
+        assertThat(getManageableMailQueue().browse())
+            .extracting(ManageableMailQueue.MailQueueItemView::getMail)
+            .extracting(Mail::getName)
+            .containsExactly(NAME1);
+    }
+
+    @Test
+    default void removeByNameShouldNotFailWhenQueueIsEmpty() throws Exception {
+        getManageableMailQueue().remove(ManageableMailQueue.Type.Name, "NAME2");
```

Comment:
oh a lost "NAME2".

Reply:
cf antoine commits, this is no more relevant